File: doc/cli/examples/Core/Base/CxException_01.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace User
{
	partial class Examples
	{
		[XIE.CxPluginExecute]
		public void CxException_01()
		{
			try
			{
				throw new XIE.CxException(XIE.ExStatus.InvalidParam);
			}
			catch (XIE.CxException ex)
			{
				Console.WriteLine("Code={0}", ex.Code);
				Console.WriteLine("{0}", ex.StackTrace);
			}
		}
	}
}

File: doc/cpp/examples/Core/Base/CxException_01.cpp

#include "xie_core.h"
#include <stdio.h>

namespace User
{

// ============================================================
void CxException_01()
{
	printf("%s\n", __FUNCTION__);

	try
	{
		// (1) 
		throw xie::CxException(xie::ExStatus::InvalidParam, __FUNCTION__, __FILE__, __LINE__);
	}
		// (2) 
	catch(const xie::CxException& ex)
	{
		// (3) 
		printf("%s(%d): Code=%d, Function=%s\n", ex.File(), ex.Line(), ex.Code(), ex.Function());
	}
}

}
